ibis/backends/__init__.py

```diff
@@ -767,6 +767,7 @@ class BaseBackend(abc.ABC, _FileIOHandler):
     def __init__(self, *args, **kwargs):
         self._con_args: tuple[Any] = args
         self._con_kwargs: dict[str, Any] = kwargs
+        self.can_reconnect: bool = True
```

Comment:
Would you mind making this private? I think we should be more intentional with what's a public attribute on a backend connection, and this feels more like an implementation detail to me. Can always expose it later if needed, but it's harder to take something away.

Reply:
Done.